pumps/prometheus.go

```diff
@@ -169,14 +169,32 @@ func (p *PrometheusPump) Init(conf interface{}) error {
 	}
 
 	//then we check the custom ones
+	errInitCustom := p.InitCustomMetrics()
+	if errInitCustom != nil {
+		p.log.Error("error initializing custom metrics:", errInitCustom)
+	}
+	p.log.Info("Starting prometheus listener on:", p.conf.Addr)
+
+	http.Handle(p.conf.Path, promhttp.Handler())
+
+	go func() {
+		log.Fatal(http.ListenAndServe(p.conf.Addr, nil))
+	}()
+	p.log.Info(p.GetName() + " Initialized")
+
+	return nil
+}
+
+//InitCustomMetrics initialise custom prometheus metrics based on p.conf.CustomMetrics and add them into p.allMetrics
+func (p *PrometheusPump) InitCustomMetrics() error {
 	if len(p.conf.CustomMetrics) > 0 {
 		customMetrics := []*PrometheusMetric{}
 		for i := range p.conf.CustomMetrics {
 			newMetric := &p.conf.CustomMetrics[i]
 			newMetric.aggregatedObservations = p.conf.AggregateObservations
 			errInit := newMetric.InitVec()
 			if errInit != nil {
-				p.log.Error(errInit)
+				return errInit
 			} else {
```

Comment:
A very little observation: this else statement is useless, as you're returning the error before if it exists

Reply:
solved!